### PR TITLE
feat(platforms): Add musl x86/aarch64 platforms

### PR DESCRIPTION
**Pull request by Hoverbear**
Adds musl based off #3657.

This is effectively a triage, as it's not an ideal solution and I'd like to revisit the way we do this to clean it up, but it's important we get this out soon.

Closes https://github.com/timberio/vector/issues/3813.
Closes https://github.com/timberio/vector/issues/3814.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -110,6 +110,44 @@ jobs:
...
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - run: sudo bash scripts/environment/bootstrap-ubuntu-20.04.sh
+      - run: bash scripts/environment/prepare.sh
+      - run: make slim-builds
+      - run: make build-x86_64-unknown-linux-gnu
```
Comment (Hoverbear) [inline .github/workflows/tests.yml:130]: Here you can see I made mistakes.

**Comment by Hoverbear** [comment]
This works but disables leveldb (which is struggling with libc++).

File: Cargo.toml [hunk 2]
```diff
@@ -506,3 +508,6 @@ required-features = ["kubernetes-e2e-tests"]
 
 [patch.'https://github.com/tower-rs/tower']
 tower-layer = "0.3"
+
+[patch.crates-io]
+leveldb-sys = { git = "https://github.com/jonas-schievink/leveldb-sys", branch = "musl-stdcxx" }
```
Comment (jamtur01) [inline Cargo.toml:513]: Looks like the failures are an issue loading this?

**Comment by Hoverbear** [comment]
@jamtur01 should I roll the Rust 1.46 update into a different PR?

**Comment by Hoverbear** [comment]
I've kicked off some local package runs.

**Comment by Hoverbear** [comment]
image

**Comment by jamtur01** [comment]
@Hoverbear If it works and isn't causing many complications then all good with me to be combined.

File: scripts/build-docker.sh [hunk 3]
```diff
@@ -39,7 +39,7 @@ build() {
       --platform="$PLATFORM" \
       --tag "$TAG" \
       target/artifacts \
-      -f "$DOCKERFILE" ${PUSH:+--push}
+      -f "$DOCKERFILE" "${PUSH:+--push}"
```
Comment (MOZGIII) [inline scripts/build-docker.sh:42]: This is actually a bug

**Comment by Hoverbear** [comment]
I need to figure out why this is failing on CI but not with a clean local 🤔 
image

**Comment by Hoverbear** [comment]
And the answer was "Ana tried to be too clever with ENV vars too soon"

**Comment by Hoverbear** [comment]
But why 
image

**Review by jamtur01 — APPROVED** [review]
I am happy if you're happy @Hoverbear

**Comment by jamtur01** [comment]
Woot!

**Comment by jamtur01** [comment]
Very big woot!

**Comment by binarylogic** [comment]
Very nice.

**Pull request merged**